### PR TITLE
Fixed db connect and connectString issues 

### PR DESCRIPTION
**Pull request by arvindgh**
Added PropertyChanged to db to split tns, Cleared param value after unchecking Encrypt, Invalid userid or password error to user, hide db password from log

### CODE REVIEW

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity decreasing per file
==============================
+ Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  -3
+ Ginger/Ginger/Environments/AppGeneralParamsPage.xaml.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5059209&bid=16651668)

File: Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs [hunk 1]
```diff
@@ -236,15 +227,34 @@ private void AddNewDB(object sender, RoutedEventArgs e)
...
             grdAppDbs.DataSourceList = AppOwner.Dbs;
         }
         #endregion Functions
 
+        private void db_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == Database.Fields.TNS)
+            {
+                Database db = (Database)sender;
+                //string intialValue = db.TNS;
```
Comment (ambarkhan) [inline Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5059209)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  -3
+ Ginger/Ginger/Environments/AppGeneralParamsPage.xaml.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5059209&bid=16651668)

**Comment by arvindgh** [comment]
tested for string variables working fine

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  -3
+ Ginger/Ginger/Environments/AppGeneralParamsPage.xaml.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5059209&bid=16651668)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  -3
+ Ginger/Ginger/Environments/AppGeneralParamsPage.xaml.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5059209&bid=16651668)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  -3
+ Ginger/Ginger/Environments/AppGeneralParamsPage.xaml.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5059209&bid=16651668)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs  -3
+ Ginger/Ginger/Environments/AppGeneralParamsPage.xaml.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5059209&bid=16651668)

**Pull request merged**